traitsui/wx/range_editor.py

```diff
@@ -74,7 +74,7 @@ class SimpleSliderEditor(BaseRangeEditor):
     #: High value for the slider range
     high = Any()
 
-    #: Formatting string used to format value and labels
+    #: Deprecated: This trait is no longer used.  
```

Comment:
Rather than simply commenting on the deprecation, I would like to make this a property which raises a deprecation warning. 
Ideally, getting the property would return the `self.factory.format_str` trait, and like wise setting it would set that trait.  However it feels incorrect for an editor instance to be trying to change a trait on its factory...

Also as I think more about this PR, I am worried about Hyrum's law here.  Even if the behavior change seems "more" correct, it is _possible_ that users are dependent on the current behavior.